service/resource/legacyv2/cluster_role_binding.go

```diff
@@ -39,7 +39,7 @@ func newClusterRoleBinding(customObject v1alpha1.FlannelConfig) *v1beta1.Cluster
 func newClusterRoleBindingForDeletion(customObject v1alpha1.FlannelConfig) *v1beta1.ClusterRoleBinding {
 	config := ClusterRoleBindingConfigDefaultConfig()
 	config.name = clusterRoleBindingForDeletion(customObject.Spec)
-	config.subjectName = clusterID(customObject.Spec)
+	config.subjectName = serviceAccountName(customObject.Spec)
```

Comment:
As they live in different namespaces better to use same service account name for both